app/recipe/tests/test_recipe_api.py

```diff
@@ -0,0 +1,207 @@
+from django.test import TestCase
+from django.urls import reverse
+
+from rest_framework.test import APIClient
+from rest_framework import status
+
+from core.models import Recipe, Ingredient
+
+RECIPE_URL = reverse('recipe:recipe-list')
+
+
+def recipe_url(id):
+    """Construct URL for a single recipe based on its ID"""
+    return reverse('recipe:recipe-detail', args=[id])
+
+
+def create_sample_recipe(**params):
+    """Helper function to create a user"""
+    return Recipe.objects.create(**params)
+
+
+class RecipeAPITests(TestCase):
+
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_create_recipe_with_ingredients(self):
+        """Test creating a recipe including ingredients"""
+        payload = {
+            'name': 'Vegan Roast Dinner',
+            'description': 'Roasted potatoes and mushroom wellington'
+                           ' with vegetables and gravy.',
+            'ingredients': [
+                {'name': 'carrots'},
+                {'name': 'potatoes'},
+                {'name': 'mushrooms'},
+            ]
+        }
+
+        response = self.client.post(RECIPE_URL, payload, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(
+            payload['name'],
+            Recipe.objects.get(id=response.data['id']).name
```

Comment:
Premature improvement here, but you could just do a `SELECT name FROM` by doing
```suggestion
            Recipe.objects.filter(id=response.data['id']).values_list("name", flat=True).get()
```